src/lambda/commands/downloadLambda.ts

```diff
@@ -154,41 +154,26 @@ async function downloadAndUnzipLambda(
         const codeLocation = response.Code!.Location!
 
         // arbitrary increments since there's no "busy" state for progress bars
-        progress.report({ increment: 10 })
+        progress.report({ message: 'Starting download' })
 
         const fetcher = new HttpResourceFetcher(codeLocation, {
-            pipeLocation: downloadLocation,
             showUrl: false,
             friendlyName: 'Lambda Function .zip file',
         })
-        await fetcher.get()
-
-        progress.report({ increment: 70 })
-
-        // HACK: `request` (currently implemented by the `fetcher.get()` call) doesn't necessarily close the pipe before returning.
-        // Brings up issues in less performant systems.
-        // keep attempting the unzip until the zip is fully built or fail after 5 seconds
-        let zipErr: Error | undefined
-        const val = await waitUntil(
-            async () => {
-                return await new Promise<boolean | undefined>(resolve => {
-                    try {
-                        new AdmZip(downloadLocation).extractAllTo(extractLocation, true)
-                        progress.report({ increment: 10 })
-                        resolve(true)
-                    } catch (err) {
-                        // err loading zip into AdmZip, prior to attempting an unzip
-                        zipErr = err as Error
-                        resolve(false)
-                    }
-                })
-            },
-            { timeout: 10000, interval: 1000, truthy: true }
-        )
+        const streams = await fetcher.get(downloadLocation)
+
+        let last: number = 0
+        streams.requestStream.on('downloadProgress', (p: Progress) => {
+            // I think these are bytes...
+            const message = p.total ? `Downloading ${p.transferred}/${p.total} bytes` : 'Downloading...'
+            const increment = p.total ? ((p.transferred - last) / p.total) * 100 : 0
+            last = p.transferred
+            progress.report({ message, increment })
+        })
 
-        if (!val) {
-            throw zipErr
-        }
+        await streams.done
+        progress.report({ message: 'Extracting...' })
```

Comment:
Nit: is it worth leaving like 5% left in the progress bar for the extracting step? Either that or maybe pop it into a separate popup?

Reply:
Oh is it getting stuck at 5% left? I didn't look too closely at the progress.

Reply:
It's moot if we turn this into a separate popup after the progress, or if we implement the progress on the fetcher level (below)...we'd still need a separate popup to signal unzipping.

Reply:
Separate pop-up could make sense. The lambda name gets in the way and hides the status messages which is unfortunate.